### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mhamidjamil**
#104 #96 these both issues are fixed in this PR

### CODE REVIEW

File: uploaded_code/uploaded_code.ino [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+// # version 5.5.1
+// # Release Note : SMS bug fix
+// module was sending sms by setting sms_allowed to true it self: logic issue
+
+=======
+//$ last work 24/Sep/23 [08:45 AM]
+// # version 5.5.0
+// # Release Note : Module can now use and update variables of SPIFFS
+// Whatsapp functionality improved
+>>>>>>> 158fb5b5088d8bedf878c206ae0b5d5e69a2e332
```
Comment (wasey-rao) [inline uploaded_code/uploaded_code.ino:12]: What is this bro? i know that this file is not useful but you also have to make sure what you upload. @mhamidjamil

File: uploaded_code/uploaded_code.ino [hunk 2]
```diff
@@ -1887,8 +1902,13 @@ void setField_MonthAndDate(int &field, int &month, int &date) {
...
                    " => field : " + String(field));
-  } else {
-    Println(7, "\tError in setField_MonthAndDate function!");
+  } else if (millis() > 20000) {
+    Println(7, "\tError in setField_MonthAndDate function!" +
+<<<<<<< HEAD
+                   String(millis() / 1000));
+=======
+                   Sting(millis() / 1000));
+>>>>>>> 158fb5b5088d8bedf878c206ae0b5d5e69a2e332
```
Comment (wasey-rao) [inline uploaded_code/uploaded_code.ino:1911]: same here

File: v5/v5_stable.ino [hunk 3]
```diff
@@ -1868,12 +1869,19 @@ bool hasPackage() {
   } else {
     Println(7, "Set Time First!");
   }
-  setField_MonthAndDate(package_expiry_date, expiryDate, expiryMonth);
-  if (RTC.month <= expiryMonth)
+  setField_MonthAndDate(package_expiry_date, expiryMonth, expiryDate);
+  if (RTC.month <= expiryMonth && RTC.month != 0) {
```
Comment (wasey-rao) [inline v5/v5_stable.ino:1873]: seems like this was the main logical issue! right?
at start they both are equal to zero and this causes issue.

**Review by wasey-rao — CHANGES_REQUESTED** [review]

File: v5/v5_stable.ino (hunk 3 above)
Comment (mhamidjamil) [reply v5/v5_stable.ino:1873]: yes, you are right.

File: uploaded_code/uploaded_code.ino (hunk 2 above)
Comment (mhamidjamil) [reply uploaded_code/uploaded_code.ino:1911]: my bad, actually i forgot 1 thing so I reset HEAD and and then make a commit which lead to this error.

**Review by wasey-rao — APPROVED** [review]

**Pull request merged**